src/ConstraintSolver.jl

```diff
@@ -146,23 +154,21 @@ mutable struct BacktrackObj
     status              :: Symbol
     variable_idx        :: Int
     pval                :: Int
-    best_bound          :: Int
+    best_bound          :: Float64
```

Comment:
Should best bound also be of type promote_T ?

Reply:
Yes thanks. I use `T` everywhere now. The user can use the option `solution_type` now for the preferred type which is supported by `MOI` but in JuMP one should probably always use `Float64` as this is the only type supported for all kind of constraints (besides `AllDifferentSet` and `EqualSet`)